packages/shared/src/lib/numberFormat.ts

```diff
@@ -13,10 +13,15 @@ export function largeNumberFormat(value: number): string | null {
   }
   if (suffixNum > 0) {
     const remainder = newValue % 1;
-    return (
-      newValue.toFixed(remainder >= 0 && remainder < 0.05 ? 0 : 1) +
-      suffixes[suffixNum]
-    );
+    // If the remainder is very small (< 0.05), show no decimal places
+    // If the remainder is significant, show one decimal place
+    const decimalPlaces = remainder >= 0 && remainder < 0.05 ? 0 : 1;
+    const formatted = newValue.toFixed(decimalPlaces);
+    // Remove trailing .0 if the number is a whole number
+    const cleanFormatted = formatted.endsWith('.0')
+      ? formatted.slice(0, -2)
+      : formatted;
+    return cleanFormatted + suffixes[suffixNum];
```

Comment:
There is a case where if we pass a value of `99,999` - the result would be `100.0K`.

The issue is that we use this function for monetary values (Cores), so I don't think we should round up which would make them think they have a higher wallet value.

Reply:
can we do this on API side, again? if it breaks we have to fix clients, on API its more easily fixable.

Reply:
I find this showing not showing of decimal places confusing, when would we show decimal places for Cores?

Reply:
> Yes, it is always integer, this is when formatted from large number, for example 10,200 which should be shown as 10.2K.

Ahh I get it, not a fan of string slicing, if we could do it mathematically with rounding I would be more confident.

Reply:
🤫